File: frontend/src/api/endpoints/economic.ts
import { apiClient } from '../client';
import type { 
  CorrelationMatrix, 
  TipAnalysis, 
  RevenueData, 
  PriceElasticity,
  SurgePricing,
  CustomerSegment,
  MarketShare,
  EconomicInsights
} from '../types';

export const getCorrelations = async () => {
  const response = await apiClient.get<CorrelationMatrix>('/economic/correlations');
  return response.data;
};

export const getTipAnalysis = async (groupBy: 'payment_type' | 'hour' | 'distance_bucket' | 'location' = 'payment_type') => {
  const response = await apiClient.get<{ data: TipAnalysis[] }>('/economic/tip-analysis', { 
    params: { group_by: groupBy } 
  });
  return response.data;
};

export const getRevenueAnalysis = async (params?: { group_by?: 'hour' | 'location' | 'zone'; year?: number }) => {
  const response = await apiClient.get<{ data: RevenueData[] }>('/economic/revenue-analysis', { params });
  return response.data;
};

export const getPriceElasticity = async () => {
  const response = await apiClient.get<{ data: PriceElasticity[] }>('/economic/price-elasticity');
  return response.data;
};

export const getSurgePricing = async (threshold: number = 3.0) => {
  const response = await apiClient.get<SurgePricing>('/economic/surge-pricing', { 
    params: { threshold } 
  });
  return response.data;
};

export const getCustomerSegments = async (nSegments: number = 5) => {
  const response = await apiClient.get<{ segments: CustomerSegment[]; n_segments: number }>('/economic/customer-segments', { 
    params: { n_segments: nSegments } 
  });
  return response.data;
};

export const getMarketShare = async () => {
  const response = await apiClient.get<{ data: MarketShare[]; total_trips: number }>('/economic/market-share');
  return response.data;
};

export const getEconomicInsights = async () => {
  const response = await apiClient.get<EconomicInsights>('/economic/insights');
  return response.data;
};

File: backend/app/api/v1/endpoints/economic.py
from fastapi import APIRouter, Query, HTTPException
from typing import Optional
from app.services.economic_service import EconomicAnalysisService

router = APIRouter()
service = EconomicAnalysisService()


@router.get("/correlations")
async def get_correlations():
    try:
        result = await service.get_correlations()
        return result
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))


@router.get("/tip-analysis")
async def get_tip_analysis(
    group_by: str = Query("payment_type", description="Group by: payment_type, hour, distance_bucket, location")
):
    try:
        result = await service.get_tip_analysis(group_by)
        return result
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))


@router.get("/revenue-analysis")
async def get_revenue_analysis(
    group_by: str = Query("hour", description="Group by: hour, location, zone"),
    year: Optional[int] = Query(None)
):
    try:
        result = await service.get_revenue_analysis(group_by, year)
        return result
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))


@router.get("/price-elasticity")
async def get_price_elasticity():
    try:
        result = await service.get_price_elasticity()
        return result
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))


@router.get("/surge-pricing")
async def get_surge_pricing(
    threshold: float = Query(3.0, description="Z-score threshold for surge detection")
):
    try:
        result = await service.get_surge_pricing(threshold)
        return result
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))


@router.get("/customer-segments")
async def get_customer_segments(
    n_segments: int = Query(5, description="Number of customer segments")
):
    try:
        result = await service.get_customer_segments(n_segments)
        return result
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))


@router.get("/market-share")
async def get_market_share():
    try:
        result = await service.get_market_share()
        return result
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))


@router.get("/insights")
async def get_economic_insights():
    try:
        result = await service.get_economic_insights()
        return result
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))

